docs/using-oci-volume-provisioner.md

```diff
@@ -0,0 +1,173 @@
+# OCI Volume Provisioner
+
+The OCI Volume Provisioner enables [dynamic provisioning][1] of storage
+resources when running Kubernetes on Oracle Cloud Infrastructure (OCI). It uses
+the [OCI Flexvolume Driver][2] to bind storage resources to Kubernetes Nodes.
+The volume provisioner offers support for [Block Volumes][5] and File System
+Storage volumes.
+
+
+## Deployment
+
+The oci-volume-provisioner is provided as a Kubernetes Deployment.
+
+### Prerequisites
+
+  - Install the [OCI flexvolume driver][2] if provisioning Block Storage
+    volumes.
+
+
+### Configuration
+
+You will first need to create a cloud-provider configuration file tailored to
+your OCI account and Kubernetes cluster. Please see the provider configuration
+documentation [here][5].
+
+```bash
+$ kubectl create secret generic oci-volume-provisioner \
+  -n kube-system \
+  --from-file=config.yaml=cloud-provider.yaml
+```
+
+### OCI Permissions
+
+If authenticating with OCI via an IAM user please ensure it has the following
+privileges in the OCI API by creating a [policy][6] tied to a group or user.
+
+```
+Allow group <name> to manage volumes in compartment <compartment>
+Allow group <name> to manage file-systems in compartment <compartment>
+```
+
+## Deploy the OCI Volume Provisioner
+
+First select the release to deploy.
+
+If your cluster is configured to use [RBAC][3] you will need to submit the
+following, replacing the <VERSION> placeholder with the selected version:
+
+```
+kubectl apply -f https://github.com/oracle/oci-volume-provisioner/releases/download/<VERSION>/oci-volume-provisioner-rbac.yaml
+```
+
+Deploy the volume provisioner into your Kubernetes cluster:
+
+```
+kubectl apply -f https://github.com/oracle/oci-volume-provisioner/releases/download/<VERSION>/oci-volume-provisioner.yaml
+```
+
+Lastly, verify that the oci-volume-provisioner is running in your cluster. By default it runs in the 'kube-system' namespace.
+
+```
+kubectl -n kube-system get po | grep volume-provisioner
+```
+
+## Usage
+
+In this example we'll use the OCI Volume Provisioner to create persistent
+storage for an NGINX Pod.
+
+### Create a StorageClass
+
+```bash
+$ cat <<EOF | kubectl apply -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: oci
+provisioner: oracle.com/oci
+EOF
+```
+
+### Create a PersistantVolumeClaim
+
+Next we'll create a [PersistentVolumeClaim][4] (PVC).
+
+The storageClassName must match the "oci" storage class supported by the
+provisioner.
+
+The matchLabels should contain the (shortened) Availability Domain (AD) within
+which you want to provision the volume. For example in Phoenix that might be
+`PHX-AD-1`, in Ashburn `US-ASHBURN-AD-1`, in Frankfurt `EU-FRANKFURT-1-AD-1`,
+and in London `UK-LONDON-1-AD-1`.
```

Comment:
Would be nice to have these in table format / a link to a central place where the short names of all the ADs is maintained, but I don't know of such a place ... I think this information needs to be crystal clear given that the naming conventions are so inconsistent.